src/test/transaction_tests.cpp

```diff
@@ -370,6 +369,40 @@ BOOST_AUTO_TEST_CASE(tx_invalid)
     }
 }
 
+BOOST_AUTO_TEST_CASE(tx_no_inputs)
+{
+    CMutableTransaction empty;
+
+    TxValidationState state;
+    BOOST_CHECK_MESSAGE(!CheckTransaction(CTransaction(empty), state), "Transaction with no inputs should be invalid.");
+    BOOST_CHECK(state.GetRejectReason() == "bad-txns-vin-empty");
+}
+
+BOOST_AUTO_TEST_CASE(tx_oversized)
+{
+    auto maxPayloadSize = MAX_BLOCK_WEIGHT / WITNESS_SCALE_FACTOR - 70;
```

Comment:
Would be nice to add a comment explaining the presence of this constant 70 for future reference.

Reply:
This is a good suggestion.  If the file changes again, an explanation of "70" would add value to future reviewers.

Reply:
I've calculated the maxPayloadSize by serializing the values that definitely overflows, subtracting the op_return data size from it, getting the baseline serialization size, which I can just subtract from (MAX_BLOCK_WEIGHT / WITNESS_SCALE_FACTOR) and get the threshold.